### PR TITLE
executor: support explain analyze for CTE statement

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/24112

trace memory and disk usage in CTE execution.

### Check List 

Tests 

- Unit test

### Release note 

- support explain analyze for CTE statement

### CODE REVIEW

File: executor/explain_test.go [hunk 1]
```diff
@@ -326,3 +327,20 @@ func (s *testSuite1) TestCheckActRowsWithUnistore(c *C) {
 		checkActRows(c, tk, test.sql, test.expected)
 	}
 }
+
+func (s *testSuite2) TestExplainAnalyzeCTEMemoryAndDiskInfo(c *C) {
```
Comment (wjhuang2016) [inline executor/explain_test.go:331]: How about adding a case for apply.

File: executor/explain_test.go [hunk 2]
```diff
@@ -326,3 +327,20 @@ func (s *testSuite1) TestCheckActRowsWithUnistore(c *C) {
...
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (a int)")
+	tk.MustExec("insert into t with recursive cte(a) as (select 1 union select a + 1 from cte where a < 1000) select * from cte;")
+
+	tk.MustExec("set @@tidb_mem_quota_query=10240;")
+	rows := tk.MustQuery("explain analyze with recursive cte(a) as (select 1 union select a + 1 from cte where a < 1000)" +
+		" select * from cte, t;").Rows()
+
```
Comment (crazycs520) [inline executor/explain_test.go:340]:
```suggestion
checkPlan := func(rows [][]interface{}, expectPlan string) {
		buf := bytes.NewBufferString("")
		for _, row := range rows {
			for i := range row {
				fmt.Fprintf(buf, "%v ", row[i])
			}
		}
		c.Assert(buf.String(), Matches, expectPlan, Commentf("%v not match %v", buf.String(), expectPlan))
	}
	expectedPlan := ".*HashJoin.*CTEFullScan.*TableReader.*TableFullScan.*CTE.*Projection.*Concurrency.*TableDual.*Projection.*Concurrency.*Selection.*CTETable.*"
	checkPlan(rows, expectedPlan)
	
```

Comment (crazycs520) [reply executor/explain_test.go:340]: Add test for the plan tree.

Comment (xiongjiwei) [reply executor/explain_test.go:342]: plan tree has been test in pr https://github.com/pingcap/tidb/pull/24986

File: executor/explain_test.go (hunk 1 above)
Comment (xiongjiwei) [reply executor/explain_test.go:331]: updated

**Review by wjhuang2016 — APPROVED** [review]

**Review by crazycs520 — COMMENTED** [review]
/lgtm

**Review by crazycs520 — APPROVED** [review]
/lgtm

**Comment by xiongjiwei** [comment]
@wjhuang2016 help merge

**Pull request merged**